### DISCUSSION TITLE
[Design/Antagonist] Revenant Rework

### DISCUSSION THREAD

**Discussion by Miraviel** [ParadiseSS13/Paradise#17757] [Documents]
# Introduction
_"The revenant is a spiteful spirit, whisked away from their own dimension due to unregulated scientific experiments. Long ago, it died a meaningless and lonely death - and now, it is here to invoke the same desolation in you."_

<details><summary>Foreword</summary>

### Foreword

- The player-facing flavour text may look too HRP at places, those are intended purely for roleplay-loving players.
- Any roleplay aspect of this antagonist is perfectly OK to be ignored from my side, I know we are MRP.
- Can you figure out how the revenant died? :>
</details>

<details><summary>Motivation and Goals</summary>

### Motivation
- The (I believe) untapped potential of a ghost-type antagonist on an MRP server
- The positive feedback on my small dimensional tear update
- And because @hal9000PR asked me to do rework revenant next. I think he was joking, but as always, I could not read the room online, so I did this

### Goals
- Making it a **more impactful** mid-round antagonist, with a **very low chance of roundending**
- Making it able to give short, **meaningful roleplay sessions** (with possible follow-up) for those roleplay-ly inclined, while still being mechanically relevant
- Making it **unkillable by weapons**, instead, people need to **help each other** to get rid of it
- To remove the dumb "**visit morgue**, visit xenobio, visit virology for free power" gameplay cycle; force it to interact with active players
- Give a reason for people to **visit the bar** (see crew greentexting)
- Create chaos but not mass-destruction
- **Spook**.

### Non-Goals
- Making it the new focus of the round: **it should NOT spawn when there is an active blob/terror spider threat**
- To spawn an antag that would **help other antags by default**. It _can_ interfere with secoffs, but mindshielded personnel get an advantage against it (up until level 3), see the skill Isolation.
</details>

# Skills and Stats

### Stats

<details><summary>Reality</summary>

#### 

_"The more it interacts with living beings, the more real it becomes. A ghost you don't believe in is a superstition. A ghost you fear is already part of you."_

- **Mechanics**: This acts as the revenant's experience meter. Three variables here:
   - `reality`: The level of the revenant. When increased, it unlocks a new skill.
   - `reality_rate`: The level progress. When it reaches the next threshold, it increases `reality` by one, and `reality_rate` becomes 0 again.
   - `reality_next_threshold`: The amount of `reality_rate` needed to increase `reality`.
- **Important:** This has to scale with the number of **active personnel with minds on the station**. I cannot stress this enough. Every x times the revenant gains `reality_rate` (maybe every three times?) via the skill Isolation, re-calculate its `reality_next_threshold`. This is to avoid the revenant spawning with 50 active personnel, then, in 15 minutes, when only 20 players are active, it would still need a fraction of 50 to gain a level. The skill Isolation is not spammable, but please make sure this calculation is still not taxing for the server.
</details>

<details><summary>Resonation</summary>

####  

_"The revenant's existence in our dimension is hanging by a thread. If we pulse it the right way, we can send it back to where it came from."_

- **Mechanics**: The **only** way to defeat a revenant is to use an Analyzer (or its PDA version) on it. When doing so, a frequency gets associated with the revenant for one minute. If someone pings this frequency with a remote signalling device (or its PDA version), the revenant loses one point and loses the frequency. If the revenant loses all Resonation points, it disappears.
  - **Caveat**: This can and will set off dormant remote signaling bombs. I am aware of it, it is a conscious design choice to unexpectedly wreak havoc.
- **Important 1:** The revenant should not have a frequency before getting analysed or we end up with people randomly pulsing frequencies. It should also immediately lose the associated frequency after a pulse, lest we end up with multiple people simultaneously pulsing it and instantly killing it.
- **Important 2:** Make sure the given frequency is pingable with PDA remote signallers. They cannot ping as low or high as handheld ones.
</details>

### Skills

<details><summary>Level 0 - Isolation</summary>

###  

_"His breathing became shallower by the minute. He tried to speak, but no words came out of his mouth anymore; even his own body betrayed him."_

#### Stage 1: 
- **Usage:** Click the skill, click the target person. Cannot be used on corpses and mobs without a mind.
- **Goal:** Attach oneself to a victim. After some time, gain some `reality_rate`.
- **Visible indicator:** Similar to the dimensional tear, electronic devices flicker nearby to give a clear visual indicator that the revenant is, indeed, here, targeting someone.
- **Flavour:** Flavour texts of isolation/despair happen to the victim.
- **Mechanic 1:** The victim goes temporarily blind and deaf.
- **Mechanic 2:** The victim's hands become unusable (fingers become numb). This is to rule out self-analysing and getting its frequency.
- **Mechanic 3:** The revenant gains the ability to telepathically talk with the victim.
- **Mechanic 4:** The victim's speech becomes scrambled. Telepathy cannot be used on it anymore (grey or genetics). The victim can still properly communicate with the revenant. (Make sure the translator implant cannot interfere with this.)
- **Mechanic 5:** The victim feels very cold; cold enough to burn, but not enough to suddenly die. Drasks instead feel warm.
- **Mechanic 6:** After a short delay (2-3 seconds) the revenant gets attached to the victim.

**Example flavour text (IPCs need a slightly different one):**
_"A sense of dread washes over you. You are going to DIE! You are going to die, you are going to-"
"Your hands go numb and start to shake."
"You look around, but you cannot see anyone..."
"You are alone."_

#### Stage 2:
- **Mechanic 1:** The victim's radio gets turned off to further isolate them. They had time to ask for help, not anymore. They can still emote (roleplay on MY roleplaying server?).
- **Mechanic 2:** The revenant has to stay in the body for some time, for example, 10 seconds. During this time, roleplay is encouraged, but not a necessity.

#### Stage 3:
- **Emergency eject:** Should the revenant feel threatened, it can break off the connection sooner, after max time/2 (5 seconds in this example). Should it decide to do so, the target suffers a great deal of burn damage (then jump to the pass out section.) Make this burn damage sane for drasks and IPCs, we don't want to immediately crit someone. The revenant only gets half as much `reality_rate` if it decides to do an emergency eject.
- **Mechanic 1:** Finally, the target passes out (gaining a nightmare flavour text), the revenant gains some `reality_rate`, and leaves the person. The psychic connection remains until the revenant moves or the target wakes up. This is purely for more roleplay possibility, to keep the passed out person entertained, should nobody be around to shake them up.
- **Mechanic 2:** Once the target wakes up, its headset is fixed, its blindness/deafness cured, its hands become usable.
- **Flavour 1:** Farewell flavour text here to tell them that the ghost left.
- **Flavour 2:** Optional, give a purely roleplay objective for the player, something akin to `Objective #1: This was a horrifying experience. Maybe a drink at the bar and someone to talk to would help me processing this.` This should default to greentext.

**Example flavour text - nightmare (slightly different for IPCs) - can write a few variations of this to keep the revenant fresh:**
_"As you hit the ground, you try to get on all fours, but you fall over someone.
Raising your head, you recognise them.
They are the most important person for you in the world, half of their head blasted off by an explosion.
Dust, death, and decay fill your nose.
This war never made sense.
...
...
...
You fall unconscious yet again..."_

#### Counterplay:
- This is **THE time to get a grip on the revenant.** Analyse the guy it inhabited, get its frequency.
- You can **still move**, albeit being blind. Run to others, try to communicate them via emotes for help.
- The **Chaplain** whacking a person with the Bible has a x% (50?) chance of exorcising the revenant. If it does so, the revenant is early ejected. If it inhabited the body for max time/2 already, it gains half as much `reality_rate`. The Chaplain gets the frequency of the revenant. Make sure you reset the victim's status if this happens; no sleeping, hands/eyes/headset get fixed.
- The revenant cannot inhabit **the same person twice**. If they keep emergency ejecting to play it safe, they should not be able to get the big boy skills.
- Consider making the **mindshield** an important factor here. What would it do? I recommend making the revenant unable to do an emergency eject. Make sure the revenant gets informed about the target being mindshielded, but only _after_ it inhabits them.
  - **Exception handling**: If the Chaplain exorcises a mindshielded person, the exorcism fails, but the Chaplain gets the frequency. NT mindshield do be strong.

#### Notes:
- This skill **has to be fine-tuned** a lot. It has to be long enough for the revenant to be threatened so it's not slaughter demon v2, but it has to be short enough for the revenant to be able to get levels.
- Give this **enough cooldown** for it to not be spammable. Maybe 20-30 seconds.
- Make sure only `/mob/living/carbon/human` can be targeted to avoid cyborgs, AIs, or Tom getting even more existential crisis.

</details>

<details><summary>Level 1 - Wail</summary>

###  
_"It was anything but fair. Seconds before the world went dark, he screamed for help one last time - but nobody came."_

#### Mechanics:
- **Usage:** Use Isolation on someone, then click this skill.
- **Goal:** Self-protecting ability to give the revenant a few more seconds to inhabit someone without getting analysed.
- **Mechanic 1:** While occupying someone, the revenant creates an EMP, pushing away people and disorienting them.
- **Mechanic 2:** The revenant is notified if the target has a cybernetic heart. This is for them to be able to make an informed decision, because:
- **Cardiac Arrest:** If the target had a cybernetic heart and thus suffers a cardiac arrest, the revenant is automatically ejected with "emergency eject" after max time/2. They should NOT be able to use it to do an emergency eject before the minimum time. Make sure the emergency eject's autosleep, numbing, etc, gets fixed after removing the revenant.
- **No murder:** If the occupied person is an IPC, the skill will do everything but the EMP part.
- **No spam:** This skill should have a considerable cooldown, preferably 2-3 minutes at least.

#### Counterplay:
- If the revenant does this, it is crystal clear **who is occupied** - the guy you flew away from. Try to scramble back to them and whack them with an Analyzer.
- EMPing someone **alerts everyone**, even outside its immediate vicinity. If people did not know there was a revenant on the station, now they do.
- If you have a cybernetic heart, consider not making the revenant feeling threatened enough to use this - **co-operate with it** and keep his presence a secret. If your hostage taker decides to end you, it was probably on you.

</details>

<details><summary>Level 2 - Foreboding</summary>

### 

_"The constant chatter, laughter, and life disgusted him. He turned his immense hatred towards the source that kept them alive in the unforgiving vacuum - and snapped it."_

- **Usage:** Stand on a wire, then click this skill.
- **Goal:** To create a station-wide havoc, but possibly not a round-ending one. We have seen powersinks resulting in shuttle calls, this does not intend to be one.
- **Mechanic 1:** A subtype of powersink is created on the spot. It has the same function as a regular powersink, but it needs holy water splashed on it or a Bible whack to destroy it.
- **Mechanic 2:** The revenant is _not quite there_ yet. This powersink deletes itself after some time (8-10 minutes?). Make it not too long for it to be roundending, not too short to be ignorable.
- **Mechanic 3:** If the powersink deletes itself, the revenant gets a huge boost to its `reality_rate`. If it gets destroyed by the crew, the revenant gains nothing from this skill (other than easy targets while the crew is looking for it).
- **No cheese:** This has to be on the station. Somehow please make sure the revenant cannot place it on solars or on the wire leading to toxins test site.

#### Counterplay:
- **Get searching.** You probably won't be able to tell if it is a revenant or an agent powersink, but bringing a cup of holy water with you cannot hurt.
- **Stay with others.** Hurling yourself into maintenance to find the thing, alone, while the station is in chaos - that is what the revenant wants. Don't give him easy points.
- Do the **usual stuff** you would do with regular powersinks (cutting wires to isolate it, check how much power is in a cable section, and so on).

</details>

<details><summary>Level 3 - Retribution</summary>

###

_"With each interaction, he became more and more grounded in a world he did not belong to. He wished to return, but the living was too resistant to be dragged along. Well then - they all just had to die."_

- **Usage:** Two skills here.
  - **Retribution - Search:** Click it to receive the location (and, if held by, the name of the person) of the nuclear authentication disk.
  - **Retribution - Begin:** Click the skill and click a person of your choice.
- **Goal:** To create a possible roundend threat by involving another person and to make losing the NAD an even more punishing act.
- **Mechanic 1:** The revenant attaches itself to the victim permanently, making it vulnerable for analysis should this skill get revealed. No more jumping around; this is the final stage.
- **Flavour 1:** The victim gets a message about the revenant inhabiting them. This will explain their objectives from a roleplay perspective.
- **Mechanic 2:** The victim gets two objectives:
  - `Objective #1: Destroy the station with a nuclear device.`
  - `Objective #2: Keep the revenant's presence a secret.`
- **Mechanic 3:** The nuclear authentication code is added to the victim's notes.
- **Mechanic 4:** The revenant is notified if the person has a cybernetic heart.
- **Mechanic 5:** The revenant is now able to use Wail with **a considerable cooldown**. Think about a minute(s) long cooldown. If the inhabited person has *any cybernetics* (or is an IPC), the skill loses its EMP property.
- **Mechanic 6:** The revenant and the victim have a permanent psychic connection (similar to Guardians). 

**Example flavour text:**
_"An overpowering presence weighs on your mind.
It whispers to you about immortality and unfathomable power.
All you need to do is to engulf the station in nuclear fire to get it.
You can probably get the disk, but what about the-
You suddenly find yourself knowing the activation code."_

#### Counterplay:
- There is no visual indicator for this skill - **keep the NAD safe**, and keep the NAD holder tight.
- This will likely not pose a problem if the captain has the NAD and he has **his people around him**. Now, if it was stolen or the holder is not overwatched...
- The revenant **cannot escape** anymore. If someone steals the NAD, and there was an active revenant, the crew can put two and two together and put down the person in question.
- I hope you kept the pinpointer safe.

#### Notes:
- This skill **should only become** available if 1) the revenant did not keep emergency ejecting himself 2) it is considerably late in the round. I cannot remember when the revenant can spawn, but it should after the one hour mark: even if the nuke does get set off, it should happen in the last 30 minutes.
- The only person who should be **immune to this skill** is the current holder of the pinpointer. We want to have a counterplay.
- This skill should be able to **target mindshielded personnel** without a problem. If the revenant decides to target the NAD holder, you may think the new antagonist will have an easier time, since they already have the NAD - but how often do captains wander off alone? If a captain can still pull this off, good on them. If the Blueshield suddenly guns down the captain on the bridge and disappears, good luck to them. Trust no one. Welcome to mafia games.
</details>

# Required Assets

<details><summary>Sprites</summary>

### Sprites
I can do these, but I appreciate the help.

- Skill icons
- A transparent mob icon for the ghosts/revenant to see. It should not be visible for living creatures whatsoever
</details>

<details><summary>Sound</summary>

### Sound

The heartbeat sound can be reused, but a new one would be nice.

- A low heartbeat sound for the skill Isolation, played for the victim and the revenant
- A not too unbearable screeching sound for the skill Wail, played for everyone around the target
</details>

**Comment by Qwertytoforty** [comment]
Summing up my suggestions / issues from the discord here:

Very rigid in weakness, only one way to counter it, analyzers (and bible, but only 1 person can use it, so), which can quickly destroy it, which heavily punishes the revenant for a screw up, but otherwise might never trigger. Incredibly fixed in how they must be dealt with, rather than being able to be beat up like everything else.

Does not seem to interact with the crew much, outside of the people who get bapped by the ghost. What it will come down to I fear, is a ghost posses bob. Bob is either: alseep on an operating table, of which nothing happens, in maints, of which nothing happens, in space repairing x, which nothing happens, or more or less anywhere alone, where they will not be able to signal to anyone. If they are not alone, IE, hallway somewhere, the crew will agree on some signal for people to scan them, and they will just /dance or whatever the signal is.

It escalates... strangely. It goes from possessing a few people, to a power sink, which will cause huge issues for 8 minutes if the revenant is smart, to going all the way to giving someone the magic nuclear codes. This is drastic, and can instantly lead to a round end, which seems strange for what it does at other levels. Most of the time, nothing will happen with this. However, if the blueshield is dead, or a vampire / hijacker is nabbed, or an antagonist who has the NAD is grabbed, this will almost certainly cause the round to end extremely fast. I also foresee command hiding the nuke, putting the disk in the safe, or testing each other with analyzers every 2 minutes, similar to old cling blood tests, which will get very silly.

(Add on, if this happens on a cult round, if a cultist is affected and the captain is nabbed, this will 100% lead to a nuke)

Overall, while interesting, it could use some tweaks, and I see this as being vastly different then revenant, and working better as something new. 

While both are ghosts, these have very different playstyles and weaknesses, and I think both work well, as they affect the round in different ways. 

Original revenant leads to more medical revives, gives work to chaplain and engineering, and can engage and mess with the crew and antagonists. It works well as a medium event, adding to the chaos, but not becoming the center issue, and interacts with whatever else is going on. It can mess with crew fighting a blob, it can help or hinder a traitor trying to do something, and maybe form an alliance with them, helping the traitor in exchange for souls. It also takes one ghost into one thing, which works fine for a medium event.

This proposal doesn't really give anyone a job / work, it just acts more like terror spider welding. Instead of everyone carrying a welder, everyone carries an analyzer, which for 95% of the crew they will never use, only a few members will use it on the ghost. (This also gives a side effect, that the station anomalies will almost certainly be defused if this has activated.) It doesn't interact with any other event / antagonist, apart from whoever it possesses, which is less of an alliance, more of something being forced on you, which the only thing you get out of it is a repulsing emp on a minute cooldown. It really just harms all the other events, it can't work with them, and will stop them by blowing it up. This makes it seem more like a major event then a medium one, however personally I think major events should try to get multiple ghost players involved, rather than just one.

I think, as stated above, these are 2 different things which fit 2 event types, major and medium, and as such can both exist, rather than replacing the other.
